src/internal/provider/version.go

```diff
@@ -0,0 +1,99 @@
+package provider
+
+import (
+	"fmt"
+	"log/slog"
+	"registry-stable/internal"
+	"registry-stable/internal/github"
+)
+
+var (
+	// From OpenTOFU go-releaser
+
+	goos = []string{
+		"darwin",
+		"freebsd",
+		"linux",
+		"windows",
+		"openbsd",
+		"solaris",
+	}
+
+	goarch = []string{
+		"386",
+		"amd64",
+		"arm",
+		"arm64",
+	}
+)
+
+func (p Provider) VersionFromRelease(release github.GHRelease) (*Version, error) {
+	version := internal.TrimTagPrefix(release.TagName)
+	artifactPrefix := fmt.Sprintf("%s_%s", p.RepositoryName(), version)
+
+	assets := make(map[string]string)
+	for _, asset := range release.ReleaseAssets.Nodes {
+		assets[asset.Name] = asset.DownloadURL
+	}
+
+	var ok bool
+	v := Version{Version: version}
+
+	for _, os := range goos {
+		for _, arch := range goarch {
+			target := Target{
+				OS:       os,
+				Arch:     arch,
+				Filename: fmt.Sprintf("%s_%s_%s.zip", artifactPrefix, os, arch),
+			}
+			target.DownloadURL, ok = assets[target.Filename]
+			if !ok {
+				// Artifact does not exist for this platform
+				continue
+			}
+
+			v.Targets = append(v.Targets, target)
+		}
+	}
```

Comment:
Whilst I like this approach right now, it may become a pain if a new OS or architecture comes out and we have to remember to update the registry to support it too. 

I think the old way of inferring the arch/os from the filename was a bit better at handling this situation. A perfect example is the recent-ish release of arm on darwin